ros/src/twist_controller/twist_controller.py

```diff
@@ -58,49 +69,24 @@ def control(self, current_vel, current_ang_vel, dbw_enabled, desired_linear_vel,
         sample_time = current_time - self.last_time
         self.last_time = current_time
 
-        throttle = self.throttle_controller.step(vel_error, sample_time)
-        brake = 0
-        decel = max(vel_error, self.decel_limit)
+        brake = 0.0
+        throttle = 0.0
+
         if desired_linear_vel <= rospy.get_param(
-                "/desired_speed_limit_apply_full_brakes", 1.75) \
-                and current_vel <= rospy.get_param(
-                "/current_speed_limit_apply_full_brakes", 3.50):
-            throttle = 0
-            brake = max(abs(decel)*self.total_vehicle_mass*self.wheel_radius,
-                        rospy.get_param("/torque_complete_stop", 400))
-
-            rospy.loginfo("Applying brakes (Complete Stop) [curr_vel,des_vel,"
-                          "err,"
-                          "t,b,"
-                          "s] [{0},"
-                          "{1},"
-                          "{2},"
-                          "{3},"
-                          "{4},"
-                          "{5}]".format(current_vel,
-                                        desired_linear_vel,
-                                        vel_error,
-                                        throttle,
-                                        brake,
-                                        steering))
-        elif throttle < rospy.get_param("/throttle_limit_apply_brakes",
-                                        0.1) and vel_error < 0:
-            if (abs(decel) > self.brake_deadband):
-                brake = abs(decel)*self.total_vehicle_mass*self.wheel_radius
-                rospy.loginfo("Applying brakes [curr_vel,des_vel,err,t,b,"
-                              "s] [{0},"
-                              "{1},"
-                              "{2},"
-                              "{3},"
-                              "{4},"
-                              "{5}]".format(current_vel,
-                                            desired_linear_vel,
-                                            vel_error,
-                                            throttle,
-                                            brake,
-                                            steering))
-            else:
-                rospy.loginfo("decel less than brake-deadband, ignoring "
-                              "brakes: {0}".format(abs(decel)))
+                "/complete_stop_speed", 0.0):
+            brake = rospy.get_param("/torque_complete_stop", 600)
+            throttle = 0.0
+            self.brake_controller.reset()
+            # rospy.logwarn('complete stop')
+        elif (vel_error < 0.0):
+            decel = max(vel_error, self.decel_limit)
+            decel = self.brake_controller.step(abs(decel), sample_time)
```

Comment:
<img width="1219" alt="screen shot 2018-05-03 at 3 13 40 am" src="https://user-images.githubusercontent.com/5813/39573628-e0013a0a-4e88-11e8-845b-e858697c874c.png">

With the PID Controller for brake:`brake_controller`, the graph is much smoother, here it shows how the current velocity catches up to the desired velocity.